docs/models/using.md

```diff
@@ -0,0 +1,133 @@
+
+!!! important
+    When you call `ModelLoader.setCustomMeshDefinition`, it **does not** call `ModelBakery.registerItemVariants`. For each and every MRL your `ItemMeshDefinition` can return, you must call this method in order for it to work.
+
+### Blockstate JSONs for Items
+
+Note that *items* can use *block*state JSONs. This is possible by simply passing an MRL pointing to a blockstate JSON into `ModelLoader.setCustomeModelResourceLocation` or returning it from an `ItemMeshDefinition`. Doing so allows you to take advantage of things like submodels (and combining variants if you use the Forge format) for item models. In the Forge format, this is especially useful for the default item layer model as you can use the `textures` block inside combining variant definitions to build up the layers of the model. Note that there is a major caveat. First, blockstate JSONs can only refer to models under `models/block`; they cannot see models under `models/item`. This means that you can't use `minecraft:item/generated` to get the default item transforms (proper rotation and scale). In a Forge blockstate JSON, you can use the model `minecraft:builtin/generated` and set the `transform` property to `forge:default-item` to work around this. Here's an example:
+
+```json
+{
+  "forge_marker": 1,
+  "defaults": {
+    "model": "builtin/generated",
+    "__comment": "Get Forge to inject the default rotations and scales for an item in your hand, on the ground, etc.",
+    "transform": "forge:default-item"
+  },
+  "variants": {
+    "base": {
+      "iron": {
+        "__comment": "Note that one layer0 and one layer1 texture will combine to form each variant.",
+        "textures": {
+	  "layer0": "examplemod:items/base_iron"
+        }
+      },
+      "__comment": "etc."
+    },
+    "emblem": {
+      "fire": {
+        "__comment": "Here we set only the second layer, which will be superimposed on the first layer defined by the base.",
+	"textures": {
+	  "layer1": "examplemod:items/emblem_fire"
+        }
+      },
+      "__comment": "etc."
+    }
+  }
+}
+```
+
+Now, the MRL `examplemod:item#base=iron,emblem=fire` points at this blockstate JSON, and the final model is the default item layer model with the "emblem_fire" texture superimposed on the "base_iron" texture.
+
+In a 1.9 multipart model, the transforms are inherited from the first model defined in it. Therefore, by creating a "block" model to house the item transforms, we can use the multipart blockstate format for items:
+
+`models/block/item/core.json`:
+```json
+{
+  "__comment": "Note how this is under models/block/item; one must resort to a bit of hackery.",
+  "parent": "item/generated",
+  "__comment": "etc."
+}
+```
+
+`blockstates/item.json`:
+```json
+{
+  "multipart": [
+    { "apply": "examplemod:item/core", "__comment": "Transforms come from here." },
+    { "when": { "floater": "cube" },
+      "apply": "examplemod:floating_cube",
+      "__comment": "etc." }
+  ]
+}
+```
```

Comment:
does this actually work?
1.9 multiparts dispatch on runtime based on IBlockState, *not* the MRL variant name, iirc.